core/src/main/java/com/netease/arctic/catalog/BaseArcticCatalog.java

```diff
@@ -425,6 +427,15 @@ public Transaction newCreateTableTransaction() {
     }
 
     protected void doCreateCheck() {
+      if (primaryKeySpec.primaryKeyExisted()) {
+        Set<String> pkSet = new HashSet<>(primaryKeySpec.fieldNames());
```

Comment:
Dose `primaryKeySpec.fieldNames().forEach(column->schema.findField(column).isOptional)` would be more simple?

Reply:
Has been modified.